hello-world/src/lib.rs

```diff
@@ -136,6 +145,45 @@ impl NFTamagotchi for Tamagotchi {
         }
     }
 
+    fn check_attributes(&mut self) {
+        // assert_eq!(msg::source(), program_id(), "Only contract can call this function");
+        
+        let curr_feed_level: u64 = self.calculate_curr_fed();
+        let curr_entertain_level: u64 = self.calculate_curr_entertained();
+        let curr_rest_level: u64 = self.calculate_curr_rest();
```

Comment:
You don't need to set types explicitly when they an be inferred.
```suggestion
        let curr_feed_level = self.calculate_curr_fed();
        let curr_entertain_level = self.calculate_curr_entertained();
        let curr_rest_level = self.calculate_curr_rest();
```